website/docs/cdktf/create-and-deploy/deployment-patterns.mdx

```diff
@@ -0,0 +1,175 @@
+
+To run Terraform Cloud with VCS integration you need to [check in your synthesized code]().
+
+## Managing Secrets and Sensitive Data
+
+Secrets appear in your synthesized CDKTF code when you read them directly from environment variables or from files with normal system access. This introduces risk, especially if you are checking the synthesized configuration into a version control system. To mitigate this, [use the `TerraformVariable` construct](/cdktf/concepts/variables-and-outputs#input-variables) to read secrets.
+
+```ts
+const adminPassword = new TerraformVariable(this, "adminPassword", {
+  type: "string",
+  description: "Admin password for the instance",
+  sensitive: true,
+});
+new MyResource(this, "hello", {
+  adminPassword: adminPassword.value, // use this instead of process.env.ADMIN_PASSWORD
+});
+```
+
+In the example above if we had used `adminPassword: process.env.ADMIN_PASSWORD` the admin password environment variable set at synthesization time would have been written to the `cdk.tf.json` file for your stack. This would have been a security risk.
+
+In contrast the `TerraformVariable` construct synthesizes a Terraform input. This input is not written to the `cdk.tf.json` file, so it is not a security risk. Instead it's only used at execution time by Terraform directly.
+
+To pass a [Terraform variable through environment variables](/cli/config/environment-variables#tf_var_name) you would need to set `TF_VAR_adminPasword='<your password>'` in the execution environment.
+
+## Example Workflows
+
+### Terraform Cloud
+
+This workflow is an example for the 'Run `cdktf synth` and handle the deployment seperately' method.
+
+Add a pre-commit step that runs `cdktf synth`. This creates Terraform configuration files under `cdktf.out/stacks/<stack-name>`. Then, configure the VCS integration for a Terraform cloud workspace to point to the appropriate path. You can use the same technique to deploy multiple independent stacks by configuring a Terraform Cloud workspace for each stack with the same VCS configuration, but a different target path.
+
+Refer to [UI/VCS-Driven Run Workflow](/cloud-docs/run/ui) in the Terraform Cloud documentation for details.
+
+### GitHub Actions CI and Terraform Cloud
+
+Set the `TERRAFORM_CLOUD_TOKEN` environment variable to the [Terraform Cloud API token](https://www.terraform.io/cloud-docs/users-teams-organizations/api-tokens). This allows the CDKTF stacks `RemoteBackend` to connect to Terraform Cloud.
+
+The following example shows a GitHub Actions configuration where the CDKTF application is deployed from the `main` branch and uses Terraform Cloud. This file could be stored in `.github/workflows/cdktf-deploy.yml`.
```

Comment:
```suggestion
Set the `TERRAFORM_CLOUD_TOKEN` environment variable to the [Terraform Cloud API token](https://www.terraform.io/cloud-docs/users-teams-organizations/api-tokens). This allows the CDKTF stacks `RemoteBackend` to connect to Terraform Cloud.

You must create a job like this for every stack so that you can pass the stack name into `cdktf diff <stack name>`. None of the stacks can depend on each other because Terraform cannot reason about which stacks must be launched first to respect these dependencies. 

The following example shows a GitHub Actions configuration where the CDKTF application is deployed from the `main` branch and uses Terraform Cloud. This file could be stored in `.github/workflows/cdktf-deploy.yml`.
```